pkg/mount/mountutil.go

```diff
@@ -99,24 +100,37 @@ func (m DiskMountUtil) getDeviceMountAttr(fn getMountData) (DeviceMountAttr, err
 //	returns syspath of that disk from which we can generate ndm given uuid of that disk.
 func getDiskDevPath(partition string) (string, error) {
 	// dev path be like /dev/sda4 we need to remove /dev/ from this string to get sys block path.
-	var diskName string
 	softlink := "/sys/class/block/" + partition
 	link, err := filepath.EvalSymlinks(softlink)
 	if err != nil {
 		return "", err
 	}
-	/*
-		link looks - /sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda4
-		parent disk is present after block then partition of that disk
-	*/
-	parts := strings.Split(link, "/")
+
+	parentDisk := getParentBlockDevice(link)
+	return "/dev/" + parentDisk, nil
+}
+
+// getParentBlockDevice returns the parent blockdevice of a given blockdevice by parsing the syspath
+//
+// syspath looks like - /sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda4
+// parent disk is present after block then partition of that disk
+//
+// for blockdevices that belong to the nvme subsystem, the symlink has a different format,
+// /sys/devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n1/nvme0n1p1. So we search for the nvme instance
```

Comment:
Have we checked the symlink format when there are more than 10 nvme devices?

Reply:
```
nvme0n1 -> ../../devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n1/
nvme0n10 -> ../../devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n10/
nvme0n11 -> ../../devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n11/
nvme0n12 -> ../../devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n12/
nvme0n1p1 -> ../../devices/pci0000:00/0000:00:0e.0/nvme/nvme0/nvme0n1/nvme0n1p1/
```

When more than 10 nvme devices are added it will come in the above format.